### PR TITLE
Pin transformer and optimum version

### PR DESCRIPTION
**Pull request by mszhanyi**
### Motivation and Context
To unblock whisper test failure

### CODE REVIEW

**Review by skottmckay — APPROVED** [review]

File: onnxruntime/python/tools/transformers/models/whisper/requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 torch>=1.13.0
-transformers>=4.24.0
+transformers==4.42.4
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/requirements.txt:2]:
```suggestion
transformers>=4.24.0,<= 4.42.4
```

Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/requirements.txt:2]:
```suggestion
transformers>=4.24.0,<= 4.42.4
```

File: onnxruntime/python/tools/transformers/models/whisper/requirements.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 openai-whisper>=20231117
 ffmpeg-python
 datasets
 soundfile
 librosa
-optimum
+optimum==1.21.2
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/requirements.txt:8]:
```suggestion
optimum<=1.21.2
```

**Comment by kunal-vaishnavi** [comment]
Can we keep the `transformers` and `optimum` versions within a range to allow users with older package versions to still use the existing files? We did something similar with the `transformers` version for LLaMA.

https://github.com/microsoft/onnxruntime/blob/b95aa0563ff99cfcda3f036fe11d8aeecb3539a1/onnxruntime/python/tools/transformers/models/llama/requirements.txt#L2

**Comment by mszhanyi** [comment]
> Can we keep the `transformers` and `optimum` versions within a range to allow users with older package versions to still use the existing files? We did something similar with the `transformers` version for LLaMA.
> 
> https://github.com/microsoft/onnxruntime/blob/b95aa0563ff99cfcda3f036fe11d8aeecb3539a1/onnxruntime/python/tools/transformers/models/llama/requirements.txt#L2

Let's try it in next PRs. we need to unblock other's PR request first

**Review by kunal-vaishnavi — APPROVED** [review]

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**